### PR TITLE
Only tag release containers if the name is different.

### PR DESCRIPTION
**Pull request by david-mcmahon**
Fixes docker build issue found recently when trying to build v1.7.0-alpha.3.
ref #44981

### CODE REVIEW

**Comment by caesarxuchao** [comment]
@k8s-bot kops aws e2e test this

File: build/lib/release.sh [hunk 1]
```diff
@@ -307,13 +310,16 @@ function kube::release::create_docker_images_for_server() {
...
         rm -rf ${docker_build_path}
 
-        # If we are building an official/alpha/beta release we want to keep docker images
-        # and tag them appropriately.
+        # If we are building an official/alpha/beta release we want to keep
+        # docker images and tag them appropriately.
         if [[ -n "${KUBE_DOCKER_IMAGE_TAG-}" && -n "${KUBE_DOCKER_REGISTRY-}" ]]; then
           local release_docker_image_tag="${KUBE_DOCKER_REGISTRY}/${binary_name}-${arch}:${KUBE_DOCKER_IMAGE_TAG}"
-          kube::log::status "Tagging docker image ${docker_image_tag} as ${release_docker_image_tag}"
-          "${DOCKER[@]}" rmi "${release_docker_image_tag}" 2>/dev/null || true
```
Comment (ixdy) [inline build/lib/release.sh]: oooh. I tested to make sure that `docker tag foo foo` worked fine, but I forgot that we did `rmi` first. thanks for the fix!

**Pull request merged**